app/models/application_record.rb

```diff
@@ -1,3 +1,8 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  def strip_whitespaces
+    self.service_name = service_name.strip unless !respond_to?("service_name") || service_name.nil?
```

Comment:
Do you need the nil checks here? Wouldn't that be caught by the respond_to already?

Reply:
You could maybe also have this take (*args) that you iterate over and the models could pass the fields that need stripping?

Reply:
can't pass in as a param from the before_validation, but added a `fields_to_strip` method that returns those fields for each class